spring-integration-cloudevents/src/test/java/org/springframework/integration/cloudevents/transformer/ToCloudEventTransformerTests.java

```diff
@@ -0,0 +1,330 @@
+		byte[] payload = convertPayloadToBytes(testRecord);
+		Message<byte[]> message = MessageBuilder.withPayload(payload).setHeader("test_id", "test-id")
+				.setHeader("contentType", JsonFormat.CONTENT_TYPE)
+				.build();
+		Object result = this.transformerWithNoExtensions.doTransform(message);
+
+		assertThat(result).isNotNull();
+		assertThat(result).isInstanceOf(Message.class);
+
+		Message<byte[]> resultMessage = (Message<byte[]>) result;
+		assertThat(resultMessage.getPayload()).isNotNull();
+		assertThat(new String(resultMessage.getPayload())).endsWith(new String(payload) + "}");
+	}
+
+	@Test
+	@SuppressWarnings("NullAway")
+	void emptyExtensionNames() {
+		Message<byte[]> message = createBaseMessage(PAYLOAD, JsonFormat.CONTENT_TYPE).build();
+
+		Object result = this.transformerWithNoExtensions.doTransform(message);
+		assertThat(result).isNotNull();
+		Message<?> resultMessage = (Message<?>) result;
+		assertThat(resultMessage.getPayload()).isNotNull();
+		assertThat(message.getPayload()).isEqualTo(PAYLOAD);
+
+	}
+
+	@Test
+	void noContentType() {
+		Message<byte[]> message = MessageBuilder.withPayload(PAYLOAD).build();
+		Message<?> result = this.transformerWithNoExtensions.transform(message);
+		assertThat(result.getHeaders().get("ce-datacontenttype")).isEqualTo("application/octet-stream");
+		assertThat(message.getPayload()).isEqualTo(PAYLOAD);
+
+	}
+
+	@Test
+	void noContentTypeNoFormatEnabled() {
+		Message<byte[]> message = MessageBuilder.withPayload(PAYLOAD).build();
+		assertThatThrownBy(() -> this.transformerWithNoExtensionsNoFormatEnabled.transform(message))
+				.isInstanceOf(MessageTransformationException.class)
+				.hasMessageContaining("No EventFormat found for 'application/octet-stream'");
+	}
+
+	@Test
+	@SuppressWarnings({"unchecked", "NullAway"})
+	void multipleExtensionMappings() {
+		String payload = "test message";
+		Message<byte[]> message = createBaseMessage(payload.getBytes(), "application/cloudevents+json")
+			.setHeader("correlation-id", "corr-999")
+				.setHeader(TRACE_HEADER, "trace-123")
+				.build();
+
+		Object result = this.transformerWithExtensions.doTransform(message);
+
+		assertThat(result).isNotNull();
+		Message<byte[]> resultMessage = (Message<byte[]>) result;
+
+		assertThat(resultMessage.getHeaders()).containsKeys("correlation-id");
+		assertThat(resultMessage.getHeaders().get("correlation-id")).isEqualTo("corr-999");
+		assertThat(new String(resultMessage.getPayload())).contains("\"traceId\":\"trace-123\"");
+		assertThat(new String(resultMessage.getPayload())).doesNotContain("\"spanId\":\"span-456\"",
+				"\"userId\":\"user-789\"");
+	}
+
+	@Test
+	void emptyStringPayloadHandling() {
+		Message<byte[]> message = createBaseMessage("".getBytes(), "application/cloudevents+json").build();
+		Object result = this.transformerWithNoExtensions.doTransform(message);
+
+		assertThat(result).isNotNull();
+		assertThat(result).isInstanceOf(Message.class);
```

Comment:
The `isNotNull()` is redundant.
The AsssertJ is smart enough to cover us with.
Plus, it is a fluent API, so we don't need to call `assertThat()` on the same object again: just chain assertions.